pregnancy-postpone.lua

```diff
@@ -0,0 +1,328 @@
+local help = [====[
+pregnancy-postpone
+=============
+Delays childbirths by a week or an user-definable amount of time (in ticks).
+Used as a workaround for a crash involving werebeasts giving birth. (https://www.bay12games.com/dwarves/mantisbt/view.php?id=11549)
+Can also be used simply to gather information about pregnancies in the world/playable area.
+Can act on: a selected unit, citizens on the map, non-citizens on the map, or worldwide.
+
+Examples
+=============
+"pregnancy-postpone --info-only --citizens --non-citizens"
+    Prints pregnancy information about all creatures in the playable area.
+
+"pregnancy-postpone --world --ticks 1200"
+    Postpones historical figures' childbirths worldwide by one day. (including those on your map)
+
+"pregnancy-postpone -iu"
+    Prints pregnancy information about the selected unit.
+
+"pregnancy-postpone -u"
+    Postpones selected unit's childbirth by a week. (the default amount of time)
+
+Options
+=============
+
+"-i" "--info-only"
+    Print pregnancy information without changing anything.
+
+"-s" "--silent"
+    Do not print pregnancy info.
+
+"-u" "--unit"
+    Acts only on the selected unit.
+
+"-c" "--citizens"
+    Acts on fort-controlled citizens in the playable area. (excludes animals)
+
+"-x" "--non-citizens"
+    Acts on all non-citizen units in the playable area, like animals and guests. (excludes citizens)
+
+"-w" "--world"
+    Act on the whole world. (INCLUDING your map, for now)
+
+"-t <number>" "--ticks <number>""
+    Postpone births by this many ticks.
+    If unspecified, the default value is 8400. (1 week)
+    in-game-time conversion table:
+      1 hour  = 50 ticks
+      1 day   = 1200 ticks
+      1 week  = 8400 ticks
+      1 month = 33600 ticks
+      1 year  = 403200 ticks
+
+Notes
+=============
+Short optional arguments can be chained together!
+   this command: "postpone-pregnancy --silent --citizens --ticks 2400"
+is identical to: "postpone-pregnancy -sct2400"
+Just make sure that the "-t <number>" argument comes last, if you want to use it.
+
+For best results, be sure that the game is paused while running the script!

```

Comment:
I'm trying to wrap my head around the user experience for this script. How is a player intended to interact with this script? should it be run on repeat in the background as a precaution? is the player supposed to save often and if DF crashes, load the last save and then run this script then? If it prevents a crash, what then? should the player just keep on postponing pregnancies forever?

Reply:
it's a one-time action that the user performs.
in my play-flow, i tend to save often, and if not, to rely on somewhat frequent autosaves.
so yeah in my case i just load the last save that is playable, and then postpone pregnancies once by an amount of time.

it's weird because i had two versions of the same save that would crash. 
One took about 5 minutes and the other less than a minute, in real time.
if i ran the command as soon as the game was loaded:
the 1 minute one was fixed by postponing pregnancies by 1 in-game day
the 5 minutes one was not! i had to postpone them for a week, so either the problem entity wasnt pregnant yet, or... idk, somehow it wasnt enough time then?


but- yeah. the idea is that the user would run this when they have a crashing save. 
When they do, they try postponing pregnancies by an amount of time, and if that fixes it: good!
i guess the best course of actions would be to wait for as close as possible to the crashing time before running it